src/Random123.jl

```diff
@@ -27,15 +27,38 @@ include("threefry.jl")
 export Philox2x, Philox4x
 include("philox.jl")
 
-if R123_USE_AESNI
-    include("aesni_common.jl")
-    export AESNI1x, AESNI4x
-    include("aesni.jl")
+"True when AES-NI has been enabled."
+const R123_USE_AESNI = Ref(false)
+export R123_USE_AESNI
+export AESNI1x, AESNI4x
+export ARS1x, ARS4x
+include("./aesni/module.jl")
 
-    export ARS1x, ARS4x
-    include("ars.jl")
-else
-    @warn "AES-NI is not enabled, so AESNI and ARS are not available."
+function __init__()
+
+    R123_USE_AESNI[] = try
+        cmd = Base.julia_cmd()
```

Comment:
`__init__` is for simple initialization; spawning a Julia process is not that and should not happen there. Either do this globally, and embed the use of AES-NI into the precompilation image (which should be safe, as that will not change across sessions) or make accesses to related functions lazy and have them initialize the bool upon first use.

Reply:
What do you mean by doing it globally? We used to do this in `deps/build.jl` but then decided that checking it every time the package is loaded may be better.

Reply:
https://github.com/JuliaRandom/Random123.jl/issues/3#issuecomment-1053161810

Reply:
Once, globally. `const use_aesni = ...`. That'll be embedded in your precompiled package and not be evaluated every time you import the package. It's still better than `deps/build.jl`, which will only fire once when you install the package, since precompilation images are (should be) regenerated once the environment changes.

But again, for full flexibility, use a dynamic getter (a function that returns whether to use AES-NI, computes that lazily, and caches it per session). Look at how Flux.jl decides whether to use GPU support or not, for inspiration.

Reply:
Thank you so much. This is really helpful! I'm gonna have a look when I have time.